opa/pkg/main.go

```diff
@@ -0,0 +1,263 @@
+									return nil, err
+								}
+								items := make([]interface{}, 0)
+								for _, item := range list.Items {
+									items = append(items, item.Object)
+								}
+								itemValue, err := ast.InterfaceToValue(items)
+								if err != nil {
+									return nil, err
+								}
+								return ast.NewTerm(itemValue), nil
+							}),
+					).PrepareForEval(ctx)
+					if err != nil {
+						return nil, err
+					}
+					obj.Object["parameters"] = checkInstance.Spec.Parameters
+					// TODO Find another way to get parameters in - Should they be a function or input?
+					// TODO Caching
+					evaluatedInput := rego.EvalInput(obj.Object)
+					results, err := query.Eval(ctx, evaluatedInput)
+					if err != nil {
+						return nil, err
+					}
+					newItems, err := processResults(obj, results, check, checkInstance)
+					if err != nil {
+						return nil, err
+					}
+					actionItems = append(actionItems, newItems...)
+				}
+			}
+		}
+	}
+	return actionItems, nil
+}
+
+func processResults(resource unstructured.Unstructured, results rego.ResultSet, check customCheck, checkInstance customCheckInstance) ([]ActionItem, error) {
+	actionItems := make([]ActionItem, 0)
+	instanceOutput := checkInstance.Spec.Output
+	checkOutput := check.Spec.Output
+	for _, result := range results {
+		for _, pack := range result.Bindings["results"].(map[string]interface{}) {
+			for _, outputArray := range pack.(map[string]interface{}) {
+				for _, output := range outputArray.([]interface{}) {
+					severity := instanceOutput.Severity
+					title := checkOutput.Title
+					remediation := checkOutput.Remediation
+					category := checkOutput.Category
+					if instanceOutput.Severity != nil {
+						severity = instanceOutput.Severity
+					}
+					if instanceOutput.Title != nil {
+						title = instanceOutput.Title
+					}
+					if instanceOutput.Remediation != nil {
+						remediation = instanceOutput.Remediation
+					}
+					if instanceOutput.Category != nil {
+						category = instanceOutput.Category
+					}
+					strMethod, ok := output.(string)
+					var description string
+					if ok {
+						description = fmt.Sprintf("String: %s", strMethod)
+					} else {
+						mapMethod, ok := output.(map[string]interface{})
+						if ok {
+							description = fmt.Sprintf("Map: Desc: %v", mapMethod["description"])
+							if mapMethod["severity"] != nil {
+								severityFloat, err := strconv.ParseFloat(mapMethod["severity"].(string), 64)
+								if err != nil {
+									return nil, err
+								}
+								severity = &severityFloat
+							}
+							if mapMethod["title"] != nil {
+								titleString := mapMethod["title"].(string)
+								title = &titleString
+							}
+
+							if mapMethod["remediation"] != nil {
+								remediationString := mapMethod["remediation"].(string)
+								remediation = &remediationString
+
+							}
+						} else {
+							description = fmt.Sprintf("Could not decipher output format of %+v %T", output, output)
```

Comment:
we should log this and just make the description blank. I think it'd be confusing to see this inside an action item

Reply:
This feels like basically malformed rego. I'm going to treat this as an error.